File: src/main/java/com/procurement/contracting/model/dto/CreateContractRS.java
package com.procurement.contracting.model.dto;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.procurement.contracting.model.dto.ocds.Contract;
import java.util.List;
import javax.validation.Valid;
import javax.validation.constraints.NotEmpty;
import lombok.Data;

@Data
public class CreateContractRS {

    @NotEmpty
    @Valid
    @JsonProperty("cans")
    private final List<Can> cans;

    @NotEmpty
    @Valid
    @JsonProperty("contracts")
    private final List<Contract> contracts;

    public CreateContractRS(@JsonProperty("cans") final List<Can> cans,
                            @JsonProperty("contracts") final List<Contract> contracts) {
        this.cans = cans;
        this.contracts = contracts;
    }
}

File: src/main/kotlin/com/procurement/contracting/model/dto/CreateContractRS.kt
package com.procurement.contracting.model.dto

import com.fasterxml.jackson.annotation.JsonInclude
import com.procurement.contracting.model.dto.ocds.Can
import com.procurement.contracting.model.dto.ocds.Contract

@JsonInclude(JsonInclude.Include.NON_NULL)
data class CreateContractRS(

        val cans: List<Can>?,

        val contracts: List<Contract>?
)
